distribution/std-lib/Base/src/Net/Http.enso

```diff
@@ -0,0 +1,99 @@
+from Base import all
+import Base.Net.Uri
+import Base.Net.Http.Config
+import Base.Net.Http.Request
+
+type Http
+
+    type Http
+
+    ## Create an OPTIONS request.
+    options : Request.To_Request -> Response
+
+    ## Create a GET request.
+    get : Request.To_Request -> Response
+
+    ## Create a POST request.
+    post : Request.To_Request -> Request.To_Post_Data -> Response
+
+    ## Create a PUT request.
+    put : Request.To_Request -> Request.To_Put_Data -> Response
+
+    ## Create a DELETE request.
+    delete : Request.To_Request -> Response
+
+    ## Create a TRACE request.
+    trace : Request.To_Request -> Response
+
+    ## Create a CONECT request.
+    connect : Request.To_Request -> Response
+
+    ## Create a request
+    request : Request.To_Request -> Response
+
+## Open an Http connection using `Uri`.
+
+   > Example
+     Open a connection and send a GET request.
+         Http.new "http://httpbin.org" http->
```

Comment:
I don't understand this pattern. What does `new` do that it requires `http` to be wrapped in a lambda? Surely it doesn't open the connection yet, so there's no underlying resource, so why not just return the builder?

Reply:
It was inspired by Ruby, and I got the idea that it'll manage the connection. You're right, the lambda is not necessary here :+1: